crates/cxx-qt-lib/src/types/qcolor.rs

```diff
@@ -3,69 +3,69 @@
 // SPDX-FileContributor: Gerhard de Clercq <gerhard.declercq@kdab.com>
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
+use cxx::{type_id, ExternType};
+use std::mem::MaybeUninit;
 
 #[cxx::bridge]
 mod ffi {
     unsafe extern "C++" {
         include!("cxx-qt-lib/include/qt_types.h");
 
-        type QColor;
+        type QColor = super::QColor;
 
         fn alpha(self: &QColor) -> i32;
         fn blue(self: &QColor) -> i32;
         fn green(self: &QColor) -> i32;
         fn red(self: &QColor) -> i32;
         #[rust_name = "set_alpha"]
-        fn setAlpha(self: Pin<&mut QColor>, red: i32);
+        fn setAlpha(self: &mut QColor, red: i32);
         #[rust_name = "set_blue"]
-        fn setBlue(self: Pin<&mut QColor>, blue: i32);
+        fn setBlue(self: &mut QColor, blue: i32);
         #[rust_name = "set_green"]
-        fn setGreen(self: Pin<&mut QColor>, green: i32);
+        fn setGreen(self: &mut QColor, green: i32);
         #[rust_name = "set_red"]
-        fn setRed(self: Pin<&mut QColor>, red: i32);
+        fn setRed(self: &mut QColor, red: i32);
 
         #[namespace = "rust::cxxqtlib1"]
-        #[rust_name = "qcolor_init"]
-        fn qcolorInit() -> UniquePtr<QColor>;
+        #[rust_name = "qcolor_init_default"]
+        fn qcolorInitDefault() -> QColor;
         #[namespace = "rust::cxxqtlib1"]
         #[rust_name = "qcolor_init_from_rgba"]
-        fn qcolorInitFromRgba(red: i32, green: i32, blue: i32, alpha: i32) -> UniquePtr<QColor>;
-        #[namespace = "rust::cxxqtlib1"]
-        #[rust_name = "qcolor_init_from_qcolor"]
-        fn qcolorInitFromQColor(color: &QColor) -> UniquePtr<QColor>;
+        fn qcolorInitFromRgba(red: i32, green: i32, blue: i32, alpha: i32) -> QColor;
     }
-
-    impl UniquePtr<QColor> {}
 }
 
-/// The QColorCpp class provides colors based on RGB, HSV or CMYK values.
+/// The QColor class provides colors based on RGB, HSV or CMYK values.
 ///
 /// Note that we only expose RGB methods for now.
-///
-/// Note that this is the C++ representation and QColor should be used in Rust.
-pub type QColor = ffi::QColor;
+#[derive(Clone)]
```

Comment:
If the C++ type doesn't have an explicit copy constructor are we OK to use derive(Clone) here ? And are we OK to skip the impl Drop if there is a default deconstructor ?

Reply:
I guess we'll find out :)

Reply:
valgrind is happy so far :-)

Reply:
https://en.cppreference.com/w/cpp/types/is_trivially_copyable
^^^^ See the notes section ^^^

Anything that is trivially copyable should be okay to derive(Clone), as Rust just does a memory-copy in the case of `[usize;X]`.

On that note, why are we currently checking for `is_trivially_copy_constructible` and `is_trivially_copy_assignable`, if we could just check for `is_trivially_copyable`?